ddl/column.go

```diff
@@ -1668,10 +1668,21 @@ func generateOriginDefaultValue(col *model.ColumnInfo) (interface{}, error) {
 	var err error
 	odValue := col.GetDefaultValue()
 	if odValue == nil && mysql.HasNotNullFlag(col.Flag) {
-		zeroVal := table.GetZeroValue(col)
-		odValue, err = zeroVal.ToString()
-		if err != nil {
-			return nil, errors.Trace(err)
+		switch col.Tp {
```

Comment:
I think `Set` also has this problem, could you fix it together?

Reply:
I just compared with mysql, current logic's result is same as mysql.

below is mysql result:

```
mysql> create table t (id int primary key, c int);
Query OK, 0 rows affected (0.02 sec)

mysql> insert into t (id, c) values (1, 1), (2, 2);
Query OK, 2 rows affected (0.00 sec)
Records: 2  Duplicates: 0  Warnings: 0

mysql> alter table t add column cc set('a','b','c','d') not null;
Query OK, 0 rows affected (0.05 sec)
Records: 0  Duplicates: 0  Warnings: 0

mysql> update t set c = 2 where id = 1;
Query OK, 1 row affected (0.00 sec)
Rows matched: 1  Changed: 1  Warnings: 0

mysql> select * from t;
+----+------+----+
| id | c    | cc |
+----+------+----+
|  1 |    2 |    |
|  2 |    2 |    |
+----+------+----+
2 rows in set (0.00 sec)

mysql> select * from t where cc = 0;
+----+------+----+
| id | c    | cc |
+----+------+----+
|  1 |    2 |    |
|  2 |    2 |    |
+----+------+----+
2 rows in set (0.00 sec)

mysql> select * from t where cc = 1;
Empty set (0.00 sec)

mysql> insert into t (id, c) values (3, 3);
ERROR 1364 (HY000): Field 'cc' doesn't have a default value
```

Then TiDB result:

```
mysql> create table t (id int primary key, c int);
Query OK, 0 rows affected (0.00 sec)

mysql> insert into t (id, c) values (1, 1), (2, 2);
Query OK, 2 rows affected (0.00 sec)
Records: 2  Duplicates: 0  Warnings: 0

mysql> alter table t add column cc set('a','b','c','d') not null;
Query OK, 0 rows affected (0.01 sec)

mysql> update t set c = 2 where id = 1;
Query OK, 1 row affected (0.00 sec)
Rows matched: 1  Changed: 1  Warnings: 0

mysql> select * from t;
+----+------+----+
| id | c    | cc |
+----+------+----+
|  1 |    2 |    |
|  2 |    2 |    |
+----+------+----+
2 rows in set (0.00 sec)

mysql> select * from t where cc = 0;
+----+------+----+
| id | c    | cc |
+----+------+----+
|  1 |    2 |    |
|  2 |    2 |    |
+----+------+----+
2 rows in set (0.00 sec)

mysql> select * from t where cc = 1;
Empty set (0.00 sec)

mysql> insert into t (id, c) values (3, 3);
ERROR 1364 (HY000): Field 'cc' doesn't have a default value
```

Reply:
Please add these results as a test.